File: RepoTransBench/target_projects/Python/Go/LanceGin_haishoku/public_tests/public_haishoku_edgecases_test.go
package public_tests

import (
	"reflect"
	"testing"
	"lancegin_haishoku/haishoku/alg"
)

func TestRgb2hlsBoundary(t *testing.T) {
	if !reflect.DeepEqual(alg.Rgb2hls([3]int{0, 0, 0}), [3]int{0, 0, 0}) {
		t.Errorf("Rgb2hls((0,0,0)) wrong")
	}
	if !reflect.DeepEqual(alg.Rgb2hls([3]int{255, 255, 255}), [3]int{0, 255, 0}) {
		t.Errorf("Rgb2hls((255,255,255)) wrong")
	}
}

File: RepoTransBench/target_projects/Python/Java/LanceGin_haishoku/src/test/java/com/example/public_tests/PublicHaishokuEdgecasesTest.java
package com.example.public_tests;

import com.example.haishoku.alg.Alg;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class PublicHaishokuEdgecasesTest {

    @Test
    public void testRgb2hlsBoundary() {
        int[] black = {0, 0, 0};
        int[] white = {255, 255, 255};
        int[] resBlack = Alg.rgb2hls(black);
        int[] resWhite = Alg.rgb2hls(white);
        assertArrayEquals(new int[] {0, 0, 0}, resBlack);
        assertArrayEquals(new int[] {0, 255, 0}, resWhite);
    }
}